### ISSUE TITLE
Bug: no-constant-condition flagging do IIFE

### ISSUE THREAD

**Issue by helixbass** [helixbass/eslint-plugin-coffee#43]
Looks like `coffee/no-constant-condition` is flagging eg:
```
return unless do ->
  ...
```
